hawtio-util/src/test/java/io/hawt/util/FilesTest.java

```diff
@@ -0,0 +1,301 @@
+  }
+
+  @Test
+  public void copyFileNotFoundException() throws IOException {
+    // Setup mocks
+    final File source = mock(File.class);
+    final File target = mock(File.class);
+
+    // Arrange
+    when(source.getAbsolutePath()).thenReturn("/tmp/source");
+    when(source.exists()).thenReturn(false);
+
+    // Act
+    thrown.expect(FileNotFoundException.class);
+    Files.copy(source, target);
+    // Method is not expected to return due to exception thrown
+  }
+
+  @Test
+  public void copyIOException() throws Exception, IOException {
+    // Setup mocks
+    final File source = mock(File.class);
+    final File target = mock(File.class);
+    final File targetParent = mock(File.class);
+
+    // Arrange
+    when(source.exists()).thenReturn(true);
+    when(target.exists()).thenReturn(false);
+
+    when(targetParent.getAbsolutePath()).thenReturn("/tmp/targetParent");
+    when(targetParent.exists()).thenReturn(false);
+    when(targetParent.mkdirs()).thenReturn(false);
+    when(target.getParentFile()).thenReturn(targetParent);
+
+    // Act
+    thrown.expect(IOException.class);
+    Files.copy(source, target);
+    // Method is not expected to return due to exception thrown
+  }
+
+  @PrepareForTest({IOHelper.class, Files.class})
+  @Test
+  public void copy() throws Exception, IOException {
+    // Setup mocks
+    final File source = mock(File.class);
+    final File target = mock(File.class);
+    final File targetParent = mock(File.class);
+    final FileInputStream is = mock(FileInputStream.class);
+    final FileOutputStream os = mock(FileOutputStream.class);
+    mockStatic(IOHelper.class);
+
+    // Arrange & Act
+    when(source.exists()).thenReturn(true);
+    when(target.getParentFile()).thenReturn(targetParent);
+    whenNew(FileInputStream.class).withParameterTypes(File.class).withArguments(isA(File.class)).thenReturn(is);
+    whenNew(FileOutputStream.class).withParameterTypes(File.class).withArguments(isA(File.class)).thenReturn(os);
+
+    when(target.exists()).thenReturn(true);
+    when(targetParent.exists()).thenReturn(false);
+    when(targetParent.mkdirs()).thenReturn(false);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+
+    when(target.exists()).thenReturn(false);
+    when(targetParent.mkdirs()).thenReturn(true);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+    
+    when(targetParent.exists()).thenReturn(true);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+  }
+
+  @Test
+  public void recursiveDelete() throws Exception {
+    // Setup mocks
+    final File file = mock(File.class);
+    final File file1 = mock(File.class);
+    final File file21 = mock(File.class);
+    final File file22 = mock(File.class);
+    final File file2 = mock(File.class);
+    final File file3 = mock(File.class);
```

Comment:
Could you give these names that indicate the structure a bit better?
Like: `rootDirectory`, `subDirectory`, `fileInRootDirectory`, `fileInSubDirectory1`, `fileInSubDirectory2`?
They are longer and not so easy on the eyes, but as it is I'm not sure it's easy to understand the "file structure" that's being mocked.

Reply:
Done that.